src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -306,7 +295,7 @@ internal struct Builder
             /// <summary>
             /// True if the fields of the builder are unset.
             /// </summary>
-            internal bool IsNull => _defaultType is null;
+            internal bool IsNull => _defaultType is null && _nullableAnnotation == 0 && (_extensions == null || _extensions == Extensions.Default);
```

Comment:
>0 [](start = 83, length = 1)

consider using a named entry (0 == Unknown)

Reply:
That would not tell me if we have the default (unset) value.  Only `0` does that.

Reply:
Then should the named entries be indexed (starting from 1) to separate "unset"?

---
In reply to: [260023332](https://github.com/dotnet/roslyn/pull/33648#discussion_r260023332) [](ancestors = 260023332)

Reply:
We cannot easily renumber the annotations, as they are hard-coded into expected attributes in many tests.

---
In reply to: [260027945](https://github.com/dotnet/roslyn/pull/33648#discussion_r260027945) [](ancestors = 260027945,260023332)